ddtrace/utils/formats.py

```diff
@@ -97,7 +98,25 @@ def parse_tags_str(tags_str):
     if not tags_str:
         return parsed_tags
 
-    for tag in tags_str.split(","):
+    # Identify separator based on which successfully identifies the correct
+    # number of valid tags
+    numtags = tags_str.count(":")
+    sep = (
+        ","
+        if sum(1 for t in tags_str.split(",") if ":" in t) == numtags
+        else " "
+        if sum(1 for t in tags_str.split(" ") if ":" in t) == numtags
+        else None
+    )
+
+    if not sep:
+        log.error(
+            "Malformed tag string: '%s'.",
```

Comment:
Should this have more helpful error message?

Should this raise? We only parse during startup right?

Reply:
A more descriptive error message could be good but this hard to guess at when we don't know for sure which separator the user is intending. I could change it to: `"Failed to find separator for tag string: '%s'."`. 

Being that it is internal, I think we can continue as is and allow it to catch the errors and log appropriately.  If we are to raise, I think we should take that up in a follow-up since that'll require a bit of work elsewhere.

Reply:
By more descriptive, we could also do something like:

> Failed to find separate for tag string: %r.
> Tag string must be comma or space separated
> one:value,two:value,three:value
> one:value two:value three:value

Saying comma or space separated with an example of how it is suppose to look (even if we don't use their string/values as an example) could be helpful, otherwise that error message means you have to go look up in the docs (if we wanted to add an "errors" section to the docs and have a link to what this error message means in long form... that could be fine too... ? maybe?)